### ISSUE TITLE
Serialized Lua Table Format

### ISSUE THREAD

**Issue by oeed** [oeed/CraftOS-Standards#18]
_Whilst this is a fairly well known format it is still important that these details are standardised._

The format must be readable using `textutils.unserialise` or running the file, but it doesn't necessarily have to be made using textutils.serialise (i.e. you can strip the spacing, new lines, etc.)

_These details are open for suggestion._

Extensions: `.tbl`
MIME: `text/table`

**Comment by demhydraz** [comment]
`application/table` (derived from JSON's application/json)

**Comment by viluon** [comment]
I would use `text/lua-table` or `application/lua-table`. We aren't dealing with furniture. Plus, the `*.ltbl` extension should be added to the list imo.

**Comment by demhydraz** [comment]
Well, not my fault English has so little words.

**Comment by oeed** [comment]
I'm cool with `application/lua-table`, although I'm not overly keen with `.ltbl` quite yet.

**Comment by demhydraz** [comment]
<pre>ltbl</pre> seems stupid IMO.

**Comment by lyqyd** [comment]
This would be why I'd suggest using `lua/data`.

**Comment by oeed** [comment]
I'd agree with @lyqyd re `lua/data`.

How about just `.data` for the extension too?

**Comment by ghost** [comment]
"data" doesn't seem nearly specific enough.

I think text/lua-table is pretty good. And then for an extension, it seems fairly obvious to be, but .table or .tbl seems like a good extension.

**Comment by oeed** [comment]
As mentioned by Lyqyd in #18, I think Lua/data or /table would be best. I'd also agree with .table

**Comment by ghost** [comment]
`Lua/table` with `.table` extension.

That seems like it would work quite nicely actually. :+1:

**Comment by viluon** [comment]
I'd go with both `*.tbl` and `*.table`. `Lua/table` is a good idea :+1:

**Comment by oeed** [comment]
I'm not so sure that we should be recommending two extensions for a newly proposed format. We're not using Windows 3.1 anymore :wink:

**Comment by viluon** [comment]
I don't see why not, many real life standards use multiple extensions, [JPEG](https://en.wikipedia.org/wiki/JPEG) is a good example, with `.jpg, .jpeg, .jpe
.jif, .jfif, .jfi` as extensions.

(Also, SVG allows `.svg` and `.svgz`, BMP `.bmp, .dib`)

**Comment by oeed** [comment]
Those extensions probably exist due to a _lack_ of standardisation. Having more than one extension without it being nec cedar feels like something standardisation should be preventing.

**Comment by viluon** [comment]
Uh, you are right. However, I am generally against long (full name) extensions (such as `.table`) and rather for abbreviations such as `*.tbl`. Not sure why exactly, I just think that a full name is.. unnecessary long, but having it as an alternative seems good to me. Forget about that if you think that's not a good idea though, it's just my personal preference.

**Comment by demhydraz** [comment]
I'm still pushing for `application/lua-table` extension `.tbl`.

**Comment by demhydraz** [comment]
Also how come I don't get a jazzy 'Contributor' badge? damn you github

**Comment by oeed** [comment]
Yeah I get what you mean @viluon. I think that because people will probably have preferences about we should list both but recommend table.

@demhydraz I think using the Lua category works best for us. They don't need to be identical to real life.

**Comment by viluon** [comment]
> recommend table

Recommend `*.table` @oeed? Why use the longer version? Text files have `*.txt` by default, not `*.text`. Extensions usually try to be as compact as possible. Since there's (imo) no other CC format that uses `*.tbl` or could be confused with `*.tbl` I would argue that this extension is **the** one to use. I'd allow `*.table` only as an alternative.

**Comment by oeed** [comment]
Fine, we'll _only_ use tbl

**Comment by ghost** [comment]
Unless anyone has any better ideas, `lua/table` and the extension `.tbl` are pretty good. I'm gonna start using them in my projects :P so I vote we just make that the standard.

**Comment by oeed** [comment]
Okay done. I'll create a COS now.

**Comment by viluon** [comment]
Use ["Closes #X" syntax](https://help.github.com/articles/closing-issues-via-commit-messages/) to close an issue automagically after merging a PR.

**Comment by oeed** [comment]
@viluon the issue is solved though. If we keep the issues open people will assume that it hasn't been done yet and create duplicate PRs.

**Comment by viluon** [comment]
And we'll leave those duplicate PRs until one of them is merged. Then close the duplicates and the related issue. I think that's the way to go @oeed!
